### PR TITLE
use react-router to navigate to /retire & removed navigateTo. 

### LINKED ISSUE #3466: Homepage is wrongly displayed for a short time instead of /retire, when proceeding with Test Pilot uninstallation
**[Affected versions]:**
- Firefox Nightly 61.0a1 

**[Affected Platforms]:**
- Windows
- Mac OS
- Linux

**[Prerequisites]:**
- Following 3 prefs have been set or added on a new profile: "xpinstall.signatures.required = false" "testpilot.env = dev" "extensions.webapi.testing = true".
- Test Pilot add-on was already installed on the profile.

**[Steps to reproduce]:**
1. Start Firefox browser with the profile from prerequisite and navigate to https://testpilot.dev.mozaws.net page.
2. Click the "Settings" button on top right side and chose to "Uninstall Test Pilot.
3. On the displayed dialog click the "Proceed" button and observe the page behavior.

**[Expected result]:**
- You are redirected to /retire page where a "Shutting down" animation is displayed, followed by the "Thanks" dialog.

**[Actual result]:**
- The "Homepage" is shortly displayed before you are redirected to the /retire page with the "Shutting down" animation. 

**[Notes]:**
- This also occurs on the experiment details page where the page shows like both the add-ons are not installed.
- Here is a screen recording of the issue: https://goo.gl/22Ez9j.

### PR DESCRIPTION
**Pull request by dannycoates**
fixes #3466 

also removed all `navigateTo` references since it's no longer used.

### CODE REVIEW

File: frontend/src/app/components/Header/tests.js [hunk 1]
```diff
@@ -120,14 +120,6 @@ describe("Header", () => {
           .simulate("click", mockClickEvent);
       };
 
-      it("should ping GA and show retire dialog on retire item click", () => {
-        clickItem("menuRetire");
-        expect(preventDefault.called).to.be.true;
-        expect(subject.state("showRetireDialog")).to.be.true;
-        expect(subject.find("RetireConfirmationDialog")).to.have.property("length", 1);
-        expectMenuGA("Retire");
```
Comment (dannycoates) [inline frontend/src/app/components/Header/tests.js:128]: this is already tested in RetireConfirmationDialog and the addition of react-router to that component broke this test

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3473/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/49a6c7a3a39daf2baae424a46f4358e4f3e1933c/

**Comment by MozillaTestPilotStorybook** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/3473/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/fc87df4445cdf468ddd70a35dba00e3a54c54bd8/

**Review by lmorchard — APPROVED** [review]
Works for me!

**Comment by fzzzy** [comment]
This had conflicts with cherry picking. It probably would have been better to just change the implementation of navigateTo to use react-router.

**Comment by fzzzy** [comment]
Never mind, I think this should not be cherry picked into the release. But still, not sure why navigateTo was removed.

**Comment by lmorchard** [comment]
> But still, not sure why navigateTo was removed.

I think because there was only one thing left using it, [which was also changed in this PR](https://github.com/mozilla/testpilot/pull/3473/files#diff-c474283c825f7ee187d43cf7c0b214cdL67).

**Pull request merged**